src/iohumdrum.cpp

```diff
@@ -796,14 +881,31 @@ void HumdrumInput::insertTitle(pugi::xml_node &titleStmt, const vector<HumdrumLi
         pugi::xml_node title = titleStmt.append_child("title");
         titlecount++;
         title.text().set(unescapeHtmlEntities(value).c_str());
-        if (lang) {
-            title.append_attribute("xml:lang") = language.c_str();
-            if (plang) {
-                title.append_attribute("type") = "main";
-            }
-            else {
-                title.append_attribute("type") = "translated";
-            }
+        if (key.compare(0, 3, "OTL") == 0) {
+          if (!lang || plang) {
+            title.append_attribute("type") = "main";
+          }
+          else {
+            title.append_attribute("type") = "translated";
+          }
+          if (lang) {
+              title.append_attribute("xml:lang") = language.c_str();
+          }
+        }
+        else if (key.compare(0, 3, "OTA") == 0) {
+          title.append_attribute("type") = "alternative";
+          if (lang) {
+              title.append_attribute("xml:lang") = language.c_str();
+          }
+        }
+        else if (key.compare(0, 3, "OTP") == 0) {
+          title.append_attribute("type") = "uniform";
```

Comment:
`OTP` is the "popular title", are you sure you want to call that the "uniform" title?

http://www.music-cog.ohio-state.edu/Humdrum/guide.append1.html